src/controllers/user.controller.ts

```diff
@@ -42,18 +43,48 @@ export class UserController {
   public async register(req: Request, res: Response): Promise<Response> {
     try {
       const { email, password } = req.body as UserAuthData;
-
+      let result;
       const existingUser = await userService.getUserByEmail(email);
 
       if (existingUser) {
-        return res
-          .status(StatusCodes.BAD_REQUEST)
-          .json({ errorMessage: "Account with this email already exists." });
+        if (existingUser.isDeleted) {
```

Comment:
you can flatten this nested ifs by combining their contidions into one, and get rid of else statement if it has no condition of execution.